tests/testcore/test_cluster.go

```diff
@@ -190,15 +198,38 @@ func (f *defaultPersistenceTestBaseFactory) NewTestBase(options *persistencetest
 }
 
 func NewClusterWithPersistenceTestBaseFactory(t *testing.T, options *TestClusterConfig, logger log.Logger, tbFactory PersistenceTestBaseFactory) (*TestCluster, error) {
-	clusterNo := GetFreeClusterNumber()
+	// determine number of hosts per service
+	const minNodes = 1
+	options.FrontendConfig.NumFrontendHosts = max(minNodes, options.FrontendConfig.NumFrontendHosts)
+	options.HistoryConfig.NumHistoryHosts = max(minNodes, options.HistoryConfig.NumHistoryHosts)
+	options.MatchingConfig.NumMatchingHosts = max(minNodes, options.MatchingConfig.NumMatchingHosts)
+	options.WorkerConfig.NumWorkers = max(minNodes, options.WorkerConfig.NumWorkers)
+	if options.WorkerConfig.DisableWorker {
+		options.WorkerConfig.NumWorkers = 0
+	}
+
+	// allocate ports
+	pp := temporalite.NewPortProvider()
+	hostsByProtocolByService := map[transferProtocol]map[primitives.ServiceName]static.Hosts{
+		grpcProtocol: map[primitives.ServiceName]static.Hosts{
+			primitives.FrontendService: static.Hosts{All: makeAddresses(pp, options.FrontendConfig.NumFrontendHosts)},
```

Comment:
```suggestion
			primitives.FrontendService: {All: makeAddresses(pp, options.FrontendConfig.NumFrontendHosts)},
```

Reply:
Ha, I wonder why (my) Goland doesn't highlight this as redundant.